x/mongo/driver/connstring/connstring.go

```diff
@@ -21,7 +21,24 @@ import (
 	"go.mongodb.org/mongo-driver/x/mongo/driver/wiremessage"
 )
 
-// Parse parses the provided uri and returns a URI object.
+// ParseAndValidate parses the provided uri and returns a URI object.
```

Comment:
If we're making changes, I'd prefer for this to say "ParseAndValidate parses the provided URI into a ConnString object" because the terminology and capitalization in the current docs is not correct. Similar changes required to `Parse` below.